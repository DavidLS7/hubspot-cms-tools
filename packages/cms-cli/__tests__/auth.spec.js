/*global banjo */

describe('Auth command', () => {
  it('works with help positional', async () =>
    expect(await banjo(['auth', 'help'])).toMatchSnapshot());

  it('works with help option', async () =>
    expect(await banjo(['auth', '--help'])).toMatchSnapshot());

  // TODO: Do we have a way to actually go through the full command?
  describe('personalaccesskey', () => {
    it('defaults to personalaccesskey', async () => {
      expect(await banjo(['auth'])).toMatchSnapshot();
    });

    it('respects the param', async () => {
      expect(await banjo(['auth', 'personalaccesskey'])).toMatchSnapshot();
    });
  });

  describe('oauth2', () => {
    it('works with oauth2', async () => {
      expect(await banjo(['auth', 'oauth2'])).toMatchSnapshot();
    });
  });
});
