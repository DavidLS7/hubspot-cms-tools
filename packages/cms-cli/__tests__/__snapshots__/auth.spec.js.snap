// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth command oauth2 works with oauth2 1`] = `
"? Enter the portal ID for your account (the number under the DOMAIN column at ht
tps://app.hubspot.com/myaccounts-beta ): [1A[41D[121C[1B"
`;

exports[`Auth command personalaccesskey defaults to personalaccesskey 1`] = `
"? When you're ready, we'll open a secure page in your default browser where you 
can view and copy your personal CMS access key, which you'll need to complete th
e next step.
<Press enter when you are ready to continue> [45D[45C"
`;

exports[`Auth command personalaccesskey respects the param 1`] = `
"? When you're ready, we'll open a secure page in your default browser where you 
can view and copy your personal CMS access key, which you'll need to complete th
e next step.
<Press enter when you are ready to continue> [45D[45C"
`;

exports[`Auth command works with help option 1`] = `
"banjo auth [type]

Configure authentication for a HubSpot account. Supported authentication
protocols are oauth2 and personalaccesskey.

Positionals:
  [type]  Authentication mechanism
                     [string] [choices: \\"personalaccesskey\\", \\"oauth2\\"] [default:
                                                            \\"personalaccesskey\\":
      An API Key tied to a specific user account. This is the recommended way of
                                   authenticating with local development tools.]

Options:
  --version     Show version number                                    [boolean]
  --debug, -d   set log level to debug                [boolean] [default: false]
  --help        Show help                                              [boolean]
  --config, -c  path to a config file                                   [string]
"
`;

exports[`Auth command works with help positional 1`] = `
"banjo auth [type]

Configure authentication for a HubSpot account. Supported authentication
protocols are oauth2 and personalaccesskey.

Positionals:
  [type]  Authentication mechanism
                     [string] [choices: \\"personalaccesskey\\", \\"oauth2\\"] [default:
                                                            \\"personalaccesskey\\":
      An API Key tied to a specific user account. This is the recommended way of
                                   authenticating with local development tools.]

Options:
  --version     Show version number                                    [boolean]
  --debug, -d   set log level to debug                [boolean] [default: false]
  --help        Show help                                              [boolean]
  --config, -c  path to a config file                                   [string]
"
`;
